onnxruntime/core/providers/migraphx/gpu_data_transfer.cc

```diff
@@ -24,17 +24,23 @@ common::Status GPUDataTransfer::CopyTensor(const Tensor& src, Tensor& dst) const
       // Copy only if the two addresses are different.
       if (dst_data != src_data) {
         HIP_CALL_THROW(hipMemcpy(dst_data, src_data, bytes, hipMemcpyDeviceToDevice));
+        #ifndef MIGRAPHX_STREAM_SYNC
         HIP_CALL_THROW(hipStreamSynchronize(nullptr));
+        #endif
       }
     } else {
       // copy from other CPU memory to GPU, this is blocking
       HIP_CALL_THROW(hipMemcpy(dst_data, src_data, bytes, hipMemcpyHostToDevice));
+      #ifndef MIGRAPHX_STREAM_SYNC
       HIP_CALL_THROW(hipStreamSynchronize(nullptr));  // TODO: still need stream sync? since already blocking
```

Comment:
Is the sync here significant here? The `hipMemcpy` is already blocking copy in the default stream. Sync on the default stream again should be a nop then.

Also, could you please format the code with clang-format.

Reply:
Formatting is complete

Reply:
If hipMemcpy is a blocking copy from CPU to Device already then performing another sync shouldn't be required. In the syncronous case we don't require the extra syncs since the EP does a sync each run if the stream hasn't completed